coriolis/licensing/client.py

```diff
@@ -0,0 +1,188 @@
+                resp.raise_for_status()
+
+        resp_data = resp.json()
+        if response_key:
+            if response_key not in resp_data:
+                raise ValueError(
+                    "No response key '%s' in response body: %s" % (
+                        response_key, resp_data))
+            resp_data = resp_data[response_key]
+
+        return resp_data
+
+    def _get(self, resource, response_key=None):
+        return self._do_req("GET", resource, response_key=response_key)
+
+    def _post(self, resource, body, response_key=None):
+        return self._do_req(
+            "POST", resource, body=body, response_key=response_key)
+
+    def _put(self, resource, body, response_key=None):
+        return self._do_req(
+            "PUT", resource, body=body, response_key=response_key)
+
+    def _delete(self, resource, body, response_key=None):
+        return self._do_req(
+            "DELETE", resource, body=body, response_key=response_key)
+
+    def get_licence_status(self):
+        """ Gets licence status for appliance. """
+        return self._get("/licence-status", "licence_status")
+
+    def get_licences(self):
+        """ Lists all installed licences. """
+        return self._get("/licences", response_key="licences")
+
+    def add_licence(self, licence_data):
+        """ Sends request to add licence (in .PEM format). """
+        return self._post("/licences", licence_data)
+
+    def add_reservation(self, typ, num_vms):
+        """ Creates a reservation of the given type. """
+        allowed_values = [
+            RESERVATION_TYPE_MIGRATION, RESERVATION_TYPE_REPLICA]
+        if typ not in allowed_values:
+            raise ValueError(
+                "Reservation type must be one of %s" % allowed_values)
+        return self._post(
+            "/reservations", {
+                "type": typ, "count": num_vms},
+            response_key="reservation")
+
+    def add_migrations_reservation(self, num_vms):
+        """ Creates a reservation for the given number of VM Migrations. """
+        return self.add_reservation(RESERVATION_TYPE_MIGRATION, num_vms)
+
+    def add_replicas_reservation(self, num_vms):
+        """ Creates a reservation for the given number of VM Replicas. """
+        return self.add_reservation(RESERVATION_TYPE_REPLICA, num_vms)
+
+    def get_reservations(self):
+        """ Lists all existing reservations. """
+        return self._get("/reservations", response_key="reservations")
+
+    def get_reservation(self, reservation_id):
+        """ Gets a reservation with the given ID.  """
+        return self._get(
+            "/reservations/%s" % reservation_id, response_key="reservation")
+
+    def check_reservation(self, reservation_id):
+        """ Checks the reservation with the given ID.  """
+        return self._post(
+            "/reservations/%s/check" % reservation_id, None,
+            response_key="reservation")
+
+    def delete_reservation(self, reservation_id):
```

Comment:
It may be useful to have a parameter like ```fail_on_404=False``` here. In some cases it may be desirable to fail if a reservation was expected to be there, but wasn't. Feel free to ignore this one. Just a thought.